client/components/app.jsx

```diff
@@ -18,10 +19,20 @@ class App extends React.Component {
       });
   }
 
+  getAverageGrade() {
+    const listOfStudents = this.state.grades;
+    const listOfGrades = listOfStudents.map(student => {
+      return student.grade;
+    });
+    const averageGrade = listOfGrades.reduce((sum, currentValue) => sum + currentValue, 0) / listOfGrades.length;
+    const averageGradeRounded = Math.round(averageGrade);
+    return averageGradeRounded;
+  }
+
   render() {
     return (
       <>
-        <Header />
+        <Header average={this.getAverageGrade} />
```

Comment:
Rather than pass the method and forcing the Header to call it, just call the method here and pass the return value.